internal/providers/internal/provider.go

```diff
@@ -0,0 +1,10 @@
+// Package internal contains the changes used by both internal/providers/sdkv2 and internal/providers/pluginfw packages.
+//
+// Note: This is different from internal/providers/common which contains the changes that *depends* on both:
+// internal/providers/sdkv2 and internal/providers/pluginfw packages. Whereas, internal/providers/internal package contains
+// the changes *used* by both internal/providers/sdkv2 and internal/providers/pluginfw packages.
+package internal
```

Comment:
I actually think that `common` would be a good name for this (see other comment). Wondering: any reason we don't merge this into the other `common` package (i.e. the one with the version)?

Reply:
I agree with having providers.go but I think we shouldn't have `common` package name for shared dependencies going forward since it will lead to loss of readability and it can grow up to contain more code changes. Eg: currently in root common we have reflection, retries, test utils, customise schema etc... all of which should ideally be under their separate package. 

Also from Naming convention as mentioned in go.dev: https://go.dev/blog/package-names
> "Avoid meaningless package names. Packages named util, common, or misc provide clients with no sense of what the package contains. This makes it harder for clients to use the package and makes it harder for maintainers to keep the package focused."

In the current context of plugin framework and sdkv2 -- they share the same tests that are defined in providers/common/tests in the PR so I thought of moving them to common but we can also move them to top level where we define providers.go, providers_test.go, providers_utils.go. 

Happy to go ahead with moving `internal/providers/internal` to `internal/providers/common` if we decide to move forward with that. + @mgyucht for discussion.

Regarding merging this into the other common package: 
- I think version and provider name should be part of providers package and not the root common package since providers package is more suitable to contain those changes (as they are related to it). 
- Right now version.go is updated automatically when we do the release and I think we should move version.go to providers package as well but this can be done later (+ need to test release automation isn't broken)

Reply:
update: Rest changes have been addressed except if we want to use the `common` package name, will make the change if we decide to move forward with it.

Reply:
updated to common